mla/threeml/sob_terms.py

```diff
@@ -0,0 +1,193 @@
+"""Docstring"""
+
+__author__ = 'John Evans'
+__copyright__ = 'Copyright 2021 John Evans'
+__credits__ = ['John Evans', 'Jason Fan', 'Michael Larson']
+__license__ = 'Apache License 2.0'
+__version__ = '0.0.1'
+__maintainer__ = 'John Evans'
+__email__ = 'john.evans@icecube.wisc.edu'
+__status__ = 'Development'
+
+import dataclasses
+
+import numpy as np
+from scipy.interpolate import UnivariateSpline as Spline
+
+from .. import sob_terms
+from .. import sources
+from .. import params as par
+from . import spectral
+from . import data_handlers
+
+
+@dataclasses.dataclass
+class ThreeMLPSEnergyTerm(sob_terms.SoBTerm):
+    """Docstring"""
+    _energysobhist: np.ndarray
+    _sin_dec_idx: np.ndarray
+    _log_energy_idx: np.ndarray
+
+    def update_sob_hist(

```

Comment:
all updates to the sobterm need to be done in the params object

Reply:
The main difference here is the likelihood minimization is done inside three and the update of the model is not represented by the params object. The user will interact through 3ML and will never interact directly with the params in 3ML. I think this way is acceptable because it doesn't require the factory to generate a new term and goes through the whole preprocessing procedure.